OrlandoDiaz_Ex4

od397 opened the pull request:
Here I’m attempting to make a South Park program where quotes fall from
the sky and are random each time, and if you click then Kenny dies and
text appears and thunder flashes. Unfortunately I had quite a bit of
trouble making the text appear from the array, making text move, and
having the thunder and text appear after clicking. Please respond so I
can correct this by tomorrow.

od397 committed:
OrlandoDiaz_Ex4

Here I’m attempting to make a South Park program where quotes fall from
the sky and are random each time, and if you click then Kenny dies and
text appears and thunder flashes. Unfortunately I had quite a bit of
trouble making the text appear from the array, making text move, and
having the thunder and text appear after clicking. Please respond so I
can correct this by tomorrow.

LadyK commented:
Understand the difference between declaring variables and instantiating them. In P5, give them values within functions, rather than globally. Ie, declare them globally, then give them an initial value, or load a list into them in the setup.

It looks like "Kenny" is an image, then you try to test whether or not it is true. Make it one or the other. Perhaps declare another value to test for truth. Put in console.log(myVar); to see the value of your variable and when/where it's changing.

You have to break these things down into steps. Code is iterative. There isn't a magic wand. Watch the debugging series I sent out earlier tonight. It would do you some good.

od397 committed:
OrlandoDiaz_Ex4

Here’s my update before finishing it up later tonight. I found out how
to make text and images appear after clicking my mouse! I also managed
to get some quotes consistently moving across the screen and triggering
follow up moving text. They’re all South Park quotes moving across! I
also want to make Cartman and Butters walk offscreen after Kenny dies
since they’re usually the cowardly ones while Stan and Kyle say the
famous line. I’ll also try to make the “You bastards!” text shake with
intensity. I’ll also add more quotes. Any advice on shaking text?

od397 committed:
OrlandoDiaz_Ex4

Here’s my South Park tribute!
Fun Fact: I finished this at the same time the new South Park came on!
Anyway, I have dozens of South Park quotes traveling across the screen.
If you click, Kenny gets struck by lightning, he dies. Kyle and Stan
yell their famous catchphrase, and Cartman and Butters walk away cause
they’re jerks. Stan’s text shakes, that took me a while so I consider
that a personal victory. I finally understand a lot of the logic behind
p5.js coding!

LadyK commented:
Very very nice job! Well done! It looks like you had fun with this one.
